### PR TITLE
Homepage frontend 1

### PR DESCRIPTION
**Pull request by ErikEmmanuelAnrique**
### **SUMMARY**
Added tabs navigation in Home Screen. Reverted changes in RootActivity session verification.

### **SCREENSHOTS**
Screenshot_1617118190
Screenshot_1617118197

### **TRELLO CARD**
https://trello.com/c/nBYVDTZw/15-u4-home-page-tabs-de-navegaci%C3%B3n

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/features/root/RootActivity.kt [hunk 1]
```diff
@@ -20,10 +20,10 @@ class RootActivity : WolmoActivity<ActivityBaseBinding>() {
     }
 
     private fun validateUser() {
-        if (userSession.isOngoingSession) {
-            HomeActivity.start(this)
-        } else {
+        if (userSession.email.isNullOrBlank() && userSession.password.isNullOrBlank()) {
```
Comment (jcruz-wolox) [inline app/src/main/java/ar/com/wolox/android/features/root/RootActivity.kt:23]: I think that you don't need check both email and password, just checking email should be enough.

File: app/src/main/res/layout/home_viewpager_fragment.xml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            app:layout_constraintEnd_toStartOf="@+id/woloxTextView"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="@+id/woloxTextView"
+            app:srcCompat="@drawable/wolox_logo" />
+
+        <TextView
+            android:id="@+id/woloxTextView"
+            android:layout_width="wrap_content"
+            android:layout_height="@dimen/wolox_big_spacing"
+            android:layout_marginStart="72dp"
```
Comment (jcruz-wolox) [inline app/src/main/res/layout/home_viewpager_fragment.xml]: Remove hard code dimen

File: app/src/main/res/layout/home_viewpager_fragment.xml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            android:gravity="center|start"
+            android:text="@string/wolox"
+            android:textAppearance="@style/TextAppearance.AppCompat.Large"
+            android:textSize="24sp"
```
Comment (jcruz-wolox) [inline app/src/main/res/layout/home_viewpager_fragment.xml]: Here too

File: app/src/main/res/layout/news_fragment.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <TextView
+            android:id="@+id/newsTextView"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="News"
+            android:textSize="36sp"
```
Comment (jcruz-wolox) [inline app/src/main/res/layout/news_fragment.xml]: Here too, remove hardcode dimen and add it in dimens xml file

File: app/src/main/res/layout/profile_fragment.xml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <TextView
+            android:id="@+id/profileTextView"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="Profile"
+            android:textSize="36sp"
```
Comment (jcruz-wolox) [inline app/src/main/res/layout/profile_fragment.xml]: Remove hardcode dimen

File: app/src/main/res/values/styles.xml [hunk 6]
```diff
@@ -8,4 +8,9 @@
         <item name="colorAccent">@color/colorAccent</item>
     </style>
 
+    <style name="tab_text" parent="@android:style/TextAppearance.Widget.TabWidget">
+        <item name="android:textSize">@dimen/font_small_more</item>
+    </style>
+
```
Comment (jcruz-wolox) [inline app/src/main/res/values/styles.xml:14]: Remove empty line

**Review by jcruz-wolox — CHANGES_REQUESTED** [review]

**Review by jcruz-wolox — APPROVED** [review]

File: app/src/main/java/ar/com/wolox/android/features/homepage/viewpager/HomeViewpagerPresenter.kt [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+import androidx.fragment.app.Fragment
+import ar.com.wolox.wolmo.core.presenter.BasePresenter
+import javax.inject.Inject
+
+class HomeViewpagerPresenter @Inject constructor() : BasePresenter<HomeViewpagerView>() {
+    fun onTabsClicked(fragment: Fragment) {
+    }
+    fun onProfileTabToggleFocus() {
+    }
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/features/homepage/viewpager/HomeViewpagerPresenter.kt]: You don't need these methods

File: app/src/main/java/ar/com/wolox/android/features/homepage/viewpager/ViewPagerAdapter.kt [hunk 8]
```diff
@@ -0,0 +1,26 @@
+package ar.com.wolox.android.features.homepage.viewpager
+
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentPagerAdapter
+
+class ViewPagerAdapter(supportFragmentManager: FragmentManager) : FragmentPagerAdapter(supportFragmentManager) {
+
+    private val mFragmentList = ArrayList<Fragment>()
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/features/homepage/viewpager/ViewPagerAdapter.kt]: I don't really like using hungarian notation here. Also, if you are declaring an ArrayList, I think it's better to not use the list word in the variable name. I would leave it just like `private val fragments`

File: app/src/main/java/ar/com/wolox/android/features/homepage/viewpager/ViewPagerAdapter.kt [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+    override fun getCount(): Int {
+        return mFragmentList.size
+    }
+
+    override fun getItem(position: Int): Fragment {
+        return mFragmentList[position]
+    }
+
+    override fun getPageTitle(position: Int): CharSequence? {
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/features/homepage/viewpager/ViewPagerAdapter.kt]: Do you need to override this method?

File: app/src/main/java/ar/com/wolox/android/utils/UserSession.kt [hunk 10]
```diff
@@ -40,6 +40,9 @@ class UserSession @Inject constructor(private val sharedPreferencesManager: Shar
         }
 
     var isOngoingSession: Boolean = false
+        get() = field.also {
+            field = it
+        }
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/utils/UserSession.kt]: I'm not getting this

**Review by ffasitella — CHANGES_REQUESTED** [review]
Good job! Just a few changes and we are ready to go 👍🏽

**Review by ffasitella — APPROVED** [review]
Nice! 🚀

**Pull request merged**